Frameworks/Foundation/NSString.mm

```diff
@@ -1088,18 +1090,21 @@ - (NSRange)rangeOfString:(NSString*)searchString options:(NSStringCompareOptions
         return matchedRange;
     }
 
-    if (range.length == 0 || findStringLength == 0) { // ??? This last item can't be here for correct Unicode compares
-        return NSMakeRange(NSNotFound, 0);
-    }
+    if (range.length > 0 && findStringLength > 0) {
+        if ((mask & NSLiteralSearch) == 0) {
```

Comment:
+1 Please update to use early returns, it makes it much easier to read and improves quality.